synapse/handlers/directory.py

```diff
@@ -454,6 +455,15 @@ async def edit_published_room_list(
                 # per alias creation rule?
                 raise SynapseError(403, "Not allowed to publish room")
 
+            # Check if publishing is blocked by a third party module
+            allowed_by_third_party_rules = await (
+                self.third_party_event_rules.check_room_can_be_added_to_public_rooms_directory(
+                    room_id
+                )
+            )
```

Comment:
should we not make this more generic, so that the module can check `public->private` transitions too?